cmd/soroban-rpc/internal/methods/get_version_info.go

```diff
@@ -22,33 +22,17 @@ type GetVersionInfoResponse struct {
 	ProtocolVersion    uint32 `json:"protocol_version"`     //nolint:tagliatelle
 }
 
-func NewGetVersionInfoHandler(logger *log.Entry, ledgerEntryReader db.LedgerEntryReader, ledgerReader db.LedgerReader, daemon interfaces.Daemon) jrpc2.Handler {
-	coreClient := daemon.CoreClient()
-	return handler.New(func(ctx context.Context) (GetVersionInfoResponse, error) {
-		var captiveCoreVersion string
-		info, err := coreClient.Info(ctx)
-		if err != nil {
-			logger.WithError(err).Info("error occurred while calling Info endpoint of core")
-		} else {
-			captiveCoreVersion = info.Info.Build
-		}
-
-		// Fetch Protocol version
-		var protocolVersion uint32
-		readTx, err := ledgerEntryReader.NewCachedTx(ctx)
-		if err != nil {
-			logger.WithError(err).Info("Cannot create read transaction")
-		}
-		defer func() {
-			_ = readTx.Done()
-		}()
+func NewGetVersionInfoHandler(
+	logger *log.Entry,
+	ledgerEntryReader db.LedgerEntryReader,
+	ledgerReader db.LedgerReader,
+	daemon interfaces.Daemon,
+) jrpc2.Handler {
+	core := daemon.GetCore()
```

Comment:
`core` can be `nil` here, right?

Reply:
hmm I dont think so we error out early if that happens in `daemon.go`

```
core, err := newCaptiveCore(cfg, logger)
	if err != nil {
		logger.WithError(err).Fatal("could not create captive core")
	}
```